corehq/apps/api/resources/v0_5.py

```diff
@@ -222,7 +222,12 @@ def get_resource_uri(self, bundle_or_obj=None, url_name='api_dispatch_detail'):
 
     def _update(self, bundle, user_change_logger=None):
         should_save = False
+        valid_keys = [
+            'phone_numbers', 'groups', 'email', 'password', 'user_data', 'first_name', 'last_name', 'language'
```

Comment:
should this on the model? I am thinking if the same set of fields should be allowed for update in api and HQ UI as well, so this should be central and not API specific.

Reply:
Yeah I agree it would be wise to centralize that logic. I'll look into that.

Reply:
fb5bb18dc4b515f61911bfd3e5806c3f1a8693a2 is not what Manish had in mind, but I think unifying the fields that are editable via the UI vs the api would require work that is outside the scope of this ticket. I felt moving the list of valid keys to the user model at least made it more visible. @snopoke do you have any opinions on this?

Reply:
+1 on unifying the editable fields being out of the scope of this PR.

I think if you put them like this on the user class thy will also get added to the JSON output and saved in the DB. I think just making it a constant would be sufficient.

Reply:
I actually think it makes sense to store on the resource class itself, as the set of editable fields could vary from version to version.  Like, if we make a v0.6 for this API, we might want to disallow editing your password.

Also, doesn't tastypie do this for you?  I see most of the fields in this API are defined here:
https://github.com/dimagi/commcare-hq/blob/ac088c211bfee503eb3011b0aab649fb893009c9/corehq/apps/api/resources/v0_1.py#L25-L33
You should be able to pass `readonly=True` to a field to prevent users from writing to it.  It looks like the `WebUserResource` and the older versions of both only allow `GET` requests, so it shouldn't affect those.

Reply:
Offlined with Ethan but never updated on this thread. The readonly attribute does handle this for you if using `hydrate` _and_ the django ORM. I put up a solution b3d439f4e523597a112160c9daffd0dcd44e2e4d that still makes use of the readonly attribute, but in a more custom way. More on that below.